AudibleApi.Common/Converter.cs

```diff
@@ -12,22 +17,93 @@ namespace AudibleApi.Common
 	//   var list = AudibleApi.Common.Converter.FromJson<Item[]>(json);
 	public static class Converter
 	{
-		private static JsonSerializerSettings Settings { get; } = new()
+		internal static JsonSerializerSettings Settings { get; } = new()
 		{
 			MetadataPropertyHandling = MetadataPropertyHandling.Ignore,
 			DateParseHandling = DateParseHandling.None,
 			Converters =
 			{
 				new IsoDateTimeConverter { DateTimeStyles = DateTimeStyles.AssumeUniversal },
-				new RecordConverter()
+				new RecordConverter(),
+				new DtoConverter()
 			},
 		};
 
+		public static async Task<T> ReadAsDtoAsync<T>(this HttpContent content) where T : DtoBase<T>
+		{
+			var jobj = await content.ReadAsJObjectAsync();
+			try
+			{
+				return DtoBase<T>.FromJson(jobj);
+			}
+			catch(Exception ex)
+			{
+				Serilog.Log.Logger.Error(ex, $"Error converting {typeof(T).Name}. Full json:\r\n" + jobj.ToString(Formatting.None));
```

Comment:
2 concerns: what if this is friggin huge? what if it contains sensitive info?

Reply:
Well, we've been doing it this way.

https://github.com/rmcrackan/AudibleApi/blob/46bb17c70b161bd622f36f34d746de3f22964101/AudibleApi/Api.Library.cs#L321-L331

https://github.com/rmcrackan/AudibleApi/blob/46bb17c70b161bd622f36f34d746de3f22964101/AudibleApi/Api.Library.cs#L249-L261

https://github.com/rmcrackan/AudibleApi/blob/46bb17c70b161bd622f36f34d746de3f22964101/AudibleApi/ApiUnauthenticated.Catalog.cs#L118-L128

https://github.com/rmcrackan/AudibleApi/blob/46bb17c70b161bd622f36f34d746de3f22964101/AudibleApi/ApiUnauthenticated.Content.cs#L21-L30

Reply:
Good point

Reply:
Actually, there is one place where we weren't logging the whole response: GetDownloadLicenseAsync. I'll add an overload for logging level.

Reply:
nvm. GetDownloadLicenseAsync uses its own special method for parsing to decrypt the voucher. Everywhere ReadAsDtoAsync is used is free of pii.